lib/glass_octopus/configuration.rb

```diff
@@ -1,59 +1,81 @@
-  #   a {BoundedExecutor} with 25 threads.
+  # @!attribute [r] connection_adapter
+  #   The configured connection adapter.
+  #   @see #adapter
   # @!attribute [rw] logger
   #   A standard library compatible logger for the application. By default it
   #   logs to the STDOUT.
-  # @!attribute [rw] shutdown_timeout
-  #   Number of seconds to wait for the processing to finish before shutting down.
   class Configuration
-    attr_accessor :connection_adapter,
-                  :executor,
-                  :logger,
-                  :shutdown_timeout
+    attr_accessor :logger
+    attr_reader :connection_adapter
 
     def initialize
       self.logger = Logger.new(STDOUT).tap { |l| l.level = Logger::INFO }
-      self.executor = default_executor
-      self.shutdown_timeout = 10
     end
 
-    # Creates a new adapter
+    # Configures a new adapter.
+    #
+    # When a class is passed as +type+ the class will be instantiated.
+    #
+    # @example Using a custom adapter class
+    #   config.adapter(MyAdapter) do |c|
+    #     c.bootstrap_servers = %w[localhost:9092]
+    #     c.group_id = "mygroup"
+    #     c.topic = "mytopic"
+    #   end
+    #
+    #   class MyAdapter
+    #     def initialize
+    #       @options = OpenStruct.new
+    #       yield @options
+    #     end
+    #
+    #     def fetch_message
+    #       @consumer.each do |fetched_message|
+    #         message = Message.new(
+    #           fetched_message.topic,
+    #          fetched_message.partition,
+    #          fetched_message.offset,
+    #          fetched_message.key,
+    #          fetched_message.value
+    #        )
+    #
+    #         yield message
+    #       end
+    #     end
     #
-    # @param type [:poseidon, :ruby_kafka] type of the adapter to use
+    #     def connect
+    #       # Connect to Kafka...
+    #       @consumer = ...
+    #       self
+    #     end
+    #
+    #     def close
+    #       @consumer.close
+    #     end
+    #   end
+    #
+    # @param type [:ruby_kafka, Class] type of the adapter to use
     # @yield a block to conigure the adapter
     # @yieldparam config configuration object
     #
-    # @see PoseidonAdapter
     # @see RubyKafkaAdapter
     def adapter(type, &block)
-      self.connection_adapter = build_adapter(type, &block)
-    end
-
-    # @api private
-    def default_executor
-      BoundedExecutor.new(Concurrent::FixedThreadPool.new(25), limit: 25)
+      @connection_adapter = build_adapter(type, &block)
     end
 
     # @api private
     def build_adapter(type, &block)
       case type
-      when :poseidon
-        require "glass_octopus/connection/poseidon_adapter"
-        PoseidonAdapter.new(&block)
       when :ruby_kafka
         require "glass_octopus/connection/ruby_kafka_adapter"
-        RubyKafkaAdapter.new(&block)
+        RubyKafkaAdapter.new(logger, &block)
+      when Class
+        type.new(&block)
```

Comment:
The arbitrary adapter class may want to have `logger` injected. There are probably no use cases for this but calling adapter constructors consistently between built-in and custom adapters is probably a good thing.

Reply:
I wanted to keep the surface for the adapter class as minimal as possible. The logger is available is needed. I think keeping the interface minimal is better in this case. `RubyKafkaAdapter` is not really part of the public interface. 

The following code snippet illustrates how to access the logger if needed. 

```ruby
GlassOctopus.run(app) do |config|
  config.logger = ...

  config.adapter MyAdapter do |adapter_config|
    adapter_config.logger = config.logger
  end
end
```